infrastructure/firecracker/create.go

```diff
@@ -48,13 +48,19 @@ func (p *fcProvider) Create(ctx context.Context, vm *models.MicroVM) error {
 		return fmt.Errorf("saving firecracker config: %w", err)
 	}
 
-	id := strings.ReplaceAll(vm.ID.String(), "/", "-")
-	args := []string{"--id", id, "--boot-timer"}
+	meta := &Metadata{
+		Latest: vm.Spec.Metadata,
```

Comment:
We need to base64 decode the values of the metadata items (see this [example](https://github.com/firecracker-microvm/firecracker/blob/main/resources/tests/metadata.json)). 

Currently, we are persisting the b64 values in the metadata.json which causes cloud-init to fail:
```
[    5.417989] cloud-init[695]: 2021-11-26 08:12:05,163 - util.py[WARNING]: Failed loading yaml blob. Yaml load allows (<class 'dict'>,) root types, but got str instead
[   15.611779] cloud-init[695]: 2021-11-26 08:12:15,396 - __init__.py[WARNING]: Unhandled non-multipart (text/x-not-multipart) userdata: 'b'I2Nsb3VkLWNvbmZpZwpob3N0'...'
```

And looking at the `metadata.json`:

```json
{
 "latest": {
  "meta-data": "aW5zdGFuY2VfaWQ6IG5z.......",
  "user-data": "I2Nsb3VkLWNvbmZpZwpob3N0....."
 }
}
```

Reply:
Added decoding on [SetMetadata](https://github.com/weaveworks/flintlock/pull/288/files#diff-f8700e072b7a291e8977a1f9eefdb244d72b221697c1ff02566655cb82d9f8d9R115-R124). It tried to decode all values, but if failed, it will use the original value in case we have a raw test already.

Reply:
Great, i see the change.....will test again.